datafusion/sql/src/planner.rs

```diff
@@ -336,14 +346,40 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                     Some(cte_name.clone()),
                     &mut ctes.clone(),
                     outer_query_schema,
+                    columns,
                 )?;
                 ctes.insert(cte_name, logical_plan);
             }
         }
         let plan = self.set_expr_to_plan(*set_expr, alias, ctes, outer_query_schema)?;
 
         let plan = self.order_by(plan, query.order_by)?;
-        self.limit(plan, query.offset, query.limit)
+        let plan = self.limit(plan, query.offset, query.limit);
+        match columns {
+            Some(_) => {
+                let mut new_builder =
+                    LogicalPlanBuilder::from(plan.as_ref().unwrap().clone());
+                new_builder = new_builder
+                    .project(
+                        plan.as_ref()
+                            .unwrap()
+                            .schema()
+                            .fields()
+                            .iter()
+                            .zip(columns.unwrap().into_iter())
+                            .map(|(field, ident)| {
+                                col(field.name()).alias(&normalize_ident(ident))
+                            }),
+                    )
+                    .map_err(|e| {
+                        context!("Failed to apply alias to inline projection.\n", e)
+                    })?;
+                new_builder.build()
+            }
+            None => {
+                plan
+            }
+        }
```

Comment:
We can remove the `unwrap` calls here:

```suggestion
        let plan = self.limit(plan, query.offset, query.limit)?;
        match columns {
            Some(_) => {
                let mut new_builder =
                    LogicalPlanBuilder::from(plan.clone());
                new_builder = new_builder
                    .project(
                        plan
                            .schema()
                            .fields()
                            .iter()
                            .zip(columns.unwrap().into_iter())
                            .map(|(field, ident)| {
                                col(field.name()).alias(&normalize_ident(ident))
                            }),
                    )
                    .map_err(|e| {
                        context!("Failed to apply alias to inline projection.\n", e)
                    })?;
                new_builder.build()
            }
            None => {
                Ok(plan)
            }
        }
```

Reply:
Oh that's a really nice change, using the ? operator to create a value rather than a result on return! Thanks for the note, I'll make the update.